test/Microsoft.IdentityModel.Tokens.Saml.Tests/Saml2SecurityTokenHandlerTests.Validations.cs

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.IdentityModel.TestUtils;
+using Microsoft.IdentityModel.Logging;
+using Xunit;
+
+namespace Microsoft.IdentityModel.Tokens.Saml2.Tests
+{
+    public partial class Saml2SecurityTokenHandlerTests
+    {
+        [Theory, MemberData(nameof(ValidateAudienceParameterTestCases), DisableDiscoveryEnumeration = true)]
+        public void ValidateAudienceParameters(AudienceValidationTheoryData theoryData)
+        {
+            CompareContext context = TestUtilities.WriteHeader($"{this}.ValidateAudienceParameters", theoryData);
+
+            if (theoryData.AudiencesToAdd != null)
+            {
+                foreach (string audience in theoryData.AudiencesToAdd)
+                    theoryData.ValidationParameters.ValidAudiences.Add(audience);
+            }
+
+            ValidationResult<string> result = Saml2SecurityTokenHandler.ValidateAudience(
+                theoryData.Audiences,
+                theoryData.SecurityToken,
+                theoryData.ValidationParameters,
+                theoryData.CallContext);
+
+            if (result.IsSuccess)
+            {
+                IdentityComparer.AreStringsEqual(
+                    result.UnwrapResult(),
+                    theoryData.Result.UnwrapResult(),
+                    context);
+
+                theoryData.ExpectedException.ProcessNoException(context);
+            }
+            else
+            {
+                ValidationError validationError = result.UnwrapError();
+                IdentityComparer.AreStringsEqual(
+                    validationError.FailureType.Name,
+                    theoryData.Result.UnwrapError().FailureType.Name,
+                    context);
+
+                Exception exception = validationError.GetException();
+                theoryData.ExpectedException.ProcessException(exception, context);
+            }
+
+            TestUtilities.AssertFailIfErrors(context);
+        }
+
+        public static TheoryData<AudienceValidationTheoryData> ValidateAudienceParameterTestCases
+        {
+            get
+            {
+                return new TheoryData<AudienceValidationTheoryData>
```

Comment:
Are all these tests re-testing the audience validation delegate that already has these tests?
See: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/dev/test/Microsoft.IdentityModel.Tokens.Tests/Validation/AudienceValidationResultTests.cs